db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/DeleteNode.java

```diff
@@ -325,19 +325,11 @@ else if (cursorTargetTableName != null)
                     rowIdColumn.setName(COLUMNNAME);
                 }
 
-                ColumnReference columnReference = (ColumnReference) getNodeFactory().getNode(
-                        C_NodeTypes.COLUMN_REFERENCE,
-                        rowIdColumn.getName(),
-                        null,
-                        getContextManager());
-                columnReference.setSource(rowIdColumn);
-                columnReference.setNestingLevel(targetTable.getLevel());
-                columnReference.setSourceLevel(targetTable.getLevel());
                 rowLocationColumn =
                         (ResultColumn) getNodeFactory().getNode(
                                 C_NodeTypes.RESULT_COLUMN,
                                 COLUMNNAME,
-                                columnReference,
+                                rowIdColumn,
                                 getContextManager());
```

Comment:
Why is this change needed? @ascend1 @msirek can you comment on whether this could cause issues?

Reply:
If we don't expose the `CURRENT_ROW_LOCATION_NODE` in the RCL for update/delete nodes we will not generate the corresponding `getRowLocation` in the projection expression, which means we will not be able to retrieve the rowId from the projection node.
For more information please have a look at `com.splicemachine.db.impl.sql.compile.ResultColumnList#generateEvaluatedRow`

Reply:
That makes sense, thanks!